scapy/layers/inet.py

```diff
@@ -216,6 +216,20 @@ class IPOption_Traceroute(IPOption):
                    IPField("originator_ip", "0.0.0.0")]
 
 
+class IPOption_Timestamp(IPOption):
+    name = "IP Option Timestamp"
+    optclass = 2
+    option = 4
+    fields_desc = [ _IPOption_HDR,
+                    ByteField("length", 0),
```

Comment:
I'm not sure to understand: will this get the good value? Could you use a length field instead?

Reply:
Are you referring that the length field should be calculated based of the layer length?
This should be the normal behavior, but I don't know how to calculate the value of length field based of the packet length.  Could you give me some hints, please.

Reply:
The way to think about it in Scapy is: what is this length about? Here, everything is static, except for one field: `internet_address`. So this field, is actually the length of `internet_address`, adjusted by 8 (the sum of the sizes of the other fields). The issue with this precise situation is that the conditions of `ConditionalFields` are not used when the size is computed.

Anyway, here is a proposal:
```Python
class IPOption_Timestamp(IPOption):
    name = "IP Option Timestamp"
    optclass = 2
    option = 4
    fields_desc = [_IPOption_HDR,
                   FieldLenField(
                       "length", None, fmt="B",
                       length_of="internet_address",
                       adjust=lambda p, x: x + 8 - (4 if p.flg == 0 else 0)
                   ),
                   ByteField("pointer", 9),
                   BitField("oflw", 0, 4),
                   BitEnumField("flg", 1, 4,
                                {0: "timestamp_only",
                                 1: "timestamp_and_ip_addr",
                                 3: "prespecified_ip_addr"}),
                   ConditionalField(IPField("internet_address", "0.0.0.0"),
                                    lambda pkt: pkt.flg != 0),
                   IntField('timestamp', 0)]
``` 

It may seem a bit complicated but has the huge advantage of not requiring a custom `.post_build()` method which we should really avoid, and to let the reader easily understand what this field is about.

What do you think?

Reply:
I kinda disagree TBH, `post_build` is to me cleaner than some messy `adjust` tricks